From 8ff5b7fbc0a2e1e97c3812602d449b0e547962c0 Mon Sep 17 00:00:00 2001
From: Martchus <martchus@gmx.net>
Date: Mon, 11 Jun 2018 20:14:42 +0200
Subject: [PATCH] Don't redefine NOMINMAX

In file included from ../../../src/cppunit/PlugInManager.cpp:8:
../../../include/cppunit/plugin/TestPlugIn.h:149: error: "NOMINMAX" redefined [-Werror]
 #define NOMINMAX

In file included from /usr/i686-w64-mingw32/include/c++/8.1.0/i686-w64-mingw32/bits/c++config.h:508,
                 from /usr/i686-w64-mingw32/include/c++/8.1.0/bits/stl_algobase.h:59,
                 from /usr/i686-w64-mingw32/include/c++/8.1.0/bits/stl_tree.h:63,
                 from /usr/i686-w64-mingw32/include/c++/8.1.0/set:60,
                 from ../../../include/cppunit/extensions/TestFactoryRegistry.h:11,
                 from ../../../src/cppunit/PlugInManager.cpp:6:
/usr/i686-w64-mingw32/include/c++/8.1.0/i686-w64-mingw32/bits/os_defines.h:45: note: this is the location of the previous definition
 #define NOMINMAX 1
---
 include/cppunit/plugin/TestPlugIn.h        | 1 -
 src/cppunit/Win32DynamicLibraryManager.cpp | 1 -
 2 files changed, 2 deletions(-)

diff --git a/include/cppunit/plugin/TestPlugIn.h b/include/cppunit/plugin/TestPlugIn.h
index 52b5b35..ab00c5c 100644
--- a/include/cppunit/plugin/TestPlugIn.h
+++ b/include/cppunit/plugin/TestPlugIn.h
@@ -146,7 +146,6 @@ typedef CppUnitTestPlugIn *(*TestPlugInSignature)();
 #define NOUSER
 #define NOKERNEL
 #define NOSOUND
-#define NOMINMAX
 #define BLENDFUNCTION void    // for mingw & gcc
 #include <windows.h>
 #endif
diff --git a/src/cppunit/Win32DynamicLibraryManager.cpp b/src/cppunit/Win32DynamicLibraryManager.cpp
index acadf46..f8009c0 100644
--- a/src/cppunit/Win32DynamicLibraryManager.cpp
+++ b/src/cppunit/Win32DynamicLibraryManager.cpp
@@ -8,7 +8,6 @@
 #define NOUSER
 #define NOKERNEL
 #define NOSOUND
-#define NOMINMAX
 #define BLENDFUNCTION void    // for mingw & gcc  
 #include <windows.h>
 
-- 
2.17.1

